pandas/core/arrays/base.py

```diff
@@ -580,32 +583,55 @@ def factorize(self, na_sentinel=-1):
         uniques = self._from_factorized(uniques, self)
         return labels, uniques
 
-    def repeat(self, repeats, axis=None):
-        """
-        Repeat elements of an array.
+    _extension_array_shared_docs['repeat'] = """
+        Repeat elements of a %(klass)s.
 
-        .. versionadded:: 0.24.0
+        Returns a new %(klass)s where each element of the current %(klass)s
+        is repeated consecutively a given number of times.
 
         Parameters
         ----------
-        repeats : int
```

Comment:
Note that `repeat` can be an array of ints in the numpy implementation, and this appears to work for the pandas implementation as well.  Added tests for this behavior in places it didn't previously exist.